internal/daemon/plugin/vendorplugin.go

```diff
@@ -109,6 +112,10 @@ func (g *GrpcPlugin) Start(ctx context.Context) (string, int32, error) {
 		ipPort, err := g.client.Init(ctx, &pb.InitRequest{DpuMode: g.dpuMode, DpuIdentifier: string(g.dpuIdentifier)})
 		if err != nil {
 			if strings.Contains(err.Error(), "already initialized") {
+				// VSP was already initialized, mark as initialized and return the error
+				g.initMutex.Lock()
+				g.initialized = false
```

Comment:
atomic.StoreInt32(&flag, 1)
and
val := atomic.LoadInt32(&flag)
might be better than adding a new mutex?

Reply:
- If this is a performance question, then we have to ask ourselves if saving a few nanoseconds in something that runs every second makes sense.
- If this is a readability/maintainablilty question, then we have to ask ourselves how easy it is to add more/make changes to an atomic int vs a line wrapped in a mutex.

It's easier to make changes if the code uses a mutex.

Reply:
The name of the mutex suggests that it is tied to the boolean. But I am ok with this.

Typically when mutexes are used in-line (not wrapped in a function), it makes the code harder to read.

In this case what makes the code easier to read is having a function SetInitDone(bool) where the mutex is inside the function.